sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,242 @@
+# Authors: Guillaume Lemaitre <guillaume.lemaitre@inria.fr>
+#          Andreas Mueller <amueller@ais.uni-bonn.de>
+# License: BSD 3 clause
+
+import numpy as np
+
+from ..base import BaseEstimator, RegressorMixin, clone, is_regressor
+from ..linear_model import LinearRegression
+from ..utils.fixes import signature
+from ..utils.validation import check_is_fitted, check_array
+from ._function_transformer import FunctionTransformer
+
+__all__ = ['TransformedTargetRegressor']
+
+
+class TransformedTargetRegressor(BaseEstimator, RegressorMixin):
+    """Meta-estimator to regress on a transformed target.
+
+    Useful for applying a non-linear transformation in regression
+    problems. This transformation can be given as a Transformer such as the
+    QuantileTransformer or as a function and its inverse such as ``np.log`` and
+    ``np.exp``.
+
+    The computation during ``fit`` is::
+
+        regressor.fit(X, func(y))
+
+    or::
+
+        regressor.fit(X, transformer.transform(y))
+
+    The computation during ``predict`` is::
+
+        inverse_func(regressor.predict(X))
+
+    or::
+
+        transformer.inverse_transform(regressor.predict(X))
+
+    Parameters
+    ----------
+    regressor : object, (default=LinearRegression())
+        Regressor object such as derived from ``RegressorMixin``.
+
+    transformer : object, (default=None)
+        Estimator object such as derived from ``TransformerMixin``. Cannot be
+        set at the same time as ``func`` and ``inverse_func``. If ``None`` and
+        ``func`` and ``inverse_func`` are ``None`` as well, the transformer
+        will be an identity transformer.
```

Comment:
I think there was a comment by Joel earlier to mention that this will be cloned? (and same for the estimator?)